File: 프로그래머스/Level 2/피보나치 수-solve.py
def solution(n):
    fibos = [0] * 100001
    fibos[1] = 1
    fibos[2] = 1
    
    for i in range(2, n + 1):
        fibos[i] = fibos[i - 1] + fibos[i - 2]
        
    return fibos[n] % 1234567

File: 프로그래머스/Level 2/피보나치 수-solve.java
class Solution {
    public int solution(int n) {
        long[] fiboArr = new long[n + 1];
        fiboArr[1] = 1;
        
        for (int i = 2; i <= n; i++) {
            fiboArr[i] = fiboArr[i - 1] % 1234567 + fiboArr[i - 2] % 1234567;
        }
        
        return (int) fiboArr[n] % 1234567;
    }
}